### ISSUE TITLE
ELM fails after adding new pft parameters

### ISSUE THREAD

**Issue by tanzeli1982** [E3SM-Project/E3SM#2992]
**Adding new pft parameters is now broken**. The likely reason is the existence of an inconsistency between the code and the pft parameter file in regard to the dimension of several parameters. In the parameter file, these parameters (e.g., **VMAX_NFIX**) are now defined by the **allpfts** dimension (size = 1) but in the code (_components/clm/src/main/pftvarcon.F90_) they are still in the size of mxpft. I checked one old pft parameter file in which these parameters are still defined by the pft dimension (size = 25). So they are consistent in the old version. I do not know why someone changed these parameters from arrays to scalars. But if it is necessary, we need to revise the code, for example following the definition of **KM_PTASE** in pftvarcon.F90.

**Comment by bishtgautam** [comment]
@xyk-ornl, @qzhu-lbl, FYI.

**Comment by tanzeli1982** [comment]
@bishtgautam @xyk-ornl @qzhu-lbl 
Finally, I figure out why the difference exists. Right now, different BGC schemes have different parameter files correspondingly. For standard use, it is clm_params_c180301.nc. For ECA, it is clm_params.c180713.nc. For RD, it is clm_params_c180524.nc. The Gautam's land parameter tool only generates a standard use parameter file.

**Comment by tanzeli1982** [comment]
But in my view, this divergence is a serious issue that we should raise to the higher level decision team.

**Comment by bbye** [comment]
I agree. The whole point of Gautam's tool was to unify all the parameter files. By creating new ones for each flag, we run the risk of those parameter files being updated inconsistently.

**Comment by bbye** [comment]
This issue is also the cause of issue #2995. I'm not sure of the exact nature of the differences between the parameter files, but the crop model needs the new parameter file tagged c180524. Since the crop model is embedded in ELM, the default RD and CTC biogeochemistry is what the crop model also uses. The addition of the parameter file for RD when crops are off causes problems when the crop model is on because the parameter file for crops is now out of date. The developers test didn't catch it because the compset for the crop model test needs to be updated. But I can't do that until this is also resolved. 

I don't know whether to close #2995 or not since these issues are related, but not the same.

**Comment by braghiere** [comment]
Was this fixed? Or do you know how to go around this issue?
